src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/RazorInitializeEndpoint.cs

```diff
@@ -21,6 +27,12 @@ public Task<InitializeResult> HandleRequestAsync(InitializeParams request, Razor
         capabilitiesManager.SetInitializeParams(request);
         var serverCapabilities = capabilitiesManager.GetInitializeResult();
 
+        var telemetryReporter = requestContext.GetRequiredService<ITelemetryReporter>();
```

Comment:
Why not acquire the `ITelemetryReporter` via DI like you did for `LanguageServerFeatureOptions`? You can just pull it into the constructor.